pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -87,7 +87,51 @@ let
       args = rattrs (args // { inherit finalPackage overrideAttrs; });
       #              ^^^^
 
-      overrideAttrs = f0: makeDerivationExtensible (lib.extends (lib.toExtension f0) rattrs);
+      overrideAttrs =
+        f0:
+        let
+          extends' =
+            overlay: f:
+            (
+              final:
+              let
+                prev = f final;
+                thisOverlay = overlay final prev;
+                # warnForBadVersionOverride = (thisOverlay ? version && !(thisOverlay ? src));
+                warnForBadVersionOverride = (
+                  thisOverlay ? version
+                  && !(thisOverlay ? src)
+                  && args.src ? outputHash
+                  && !(
+                    thisOverlay ? __intentionallyOverridingVersion
+                    && thisOverlay.__intentionallyOverridingVersion == true
```

Comment:
This can just be `thisOverlay.__intentionallyOverridingVersion or false`, right?

Reply:
Yup, good catch.